### DISCUSSION TITLE
Retry logic

### DISCUSSION THREAD

**Discussion by kevinsimper** [prisma/prisma#9812] [Q&A]
What is your experience with retry logic with prisma. Sometimes I get errors like P2024 "Timed out fetching a new connection from the connection pool".

If prisma gets this error, prisma will not retry. And I don't see a easy way to get prisma to easily retry.

One solution could be to wrap the whole function the logic that also runs the prisma calls. That would be the best solution but it adds complexity.

The other solution could be to wrap the prisma client in a function that retries, I can see it could be added with the middleware, but I have not seen or tried it out myself yet. https://www.prisma.io/docs/concepts/components/prisma-client/middleware

What are the community experience with retrying queries?

**Comment by janpio** [comment]
I played around with something some time ago (unsuccessfully, never finished it) but also wrote a middleware to retry requests on specific errors, maybe this can give you an idea: https://github.com/janpio/serverless-prisma/blob/52a7a8e0edc3b29000970f791014dfde368a129b/demo/serverless-prisma.ts#L73-L138

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
